addons/mail/controllers/websocket.py

```diff
@@ -0,0 +1,18 @@
+from odoo.addons.bus.controllers.websocket import WebsocketController
+from odoo.http import request, route, SessionExpiredException
+
+
+class WebsocketControllerPresence(WebsocketController):
+    @route("/websocket/update_bus_presence", type="jsonrpc", auth="public", cors="*")
+    def update_bus_presence(self, inactivity_period):
+        if "is_websocket_session" not in request.session:
+            raise SessionExpiredException()
+        request.env["ir.websocket"]._update_bus_presence(int(inactivity_period))
+        return {}
+
+    @route()
+    def peek_notifications(self, channels, last, is_first_poll=False):
+        bus_channels = request.env["ir.websocket"]._build_bus_channel_list(channels)
```

Comment:
Maybe we should provide an override point in the bus method at the correct place to avoid having to build the channel list twice.

Reply:
`_get_peek_notifications_data` that returns `_prepare_subscribe_data`? It feels very random to me.
I don't fully understand how this route is used. I understand it is used in Odoo.sh but I don't understand how.